### PR TITLE
various updates/fixes to plugins to make customization easier

### PR DESCRIPTION
**Pull request by macqueen**
these changes are necessary for moving JIRA to react (figured I'd put up smaller PRs as I go to make it easier to review)

cc @getsentry/plugins 

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4197)

### CODE REVIEW

File: src/sentry/api/serializers/models/plugin.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def serialize_field(project, plugin, field):
         'name': six.text_type(field['name']),
         'label': six.text_type(field.get('label') or field['name'].title().replace('_', ' ')),
         'type': field.get('type', 'text'),
-        'required': field.get('required', False),
+        'required': field.get('required', True),
```
Comment (macqueen) [inline src/sentry/api/serializers/models/plugin.py]: this matches django forms and the `GenericField` component

File: src/sentry/static/sentry/app/components/forms/form.jsx [hunk 2]
```diff
@@ -31,6 +32,7 @@ const Form = React.createClass({
           <button className="btn btn-primary"
                   disabled={this.props.submitDisabled}
                   type="submit">{this.props.submitLabel}</button>
+          {this.props.extraButton}
```
Comment (macqueen) [inline src/sentry/static/sentry/app/components/forms/form.jsx]: this just makes something like this possible:
screenshot 2016-09-21 17 26 03

**Comment by dcramer** [comment]
per convo last night, I'm ok making required the default, but we should audit existing usage (and probably just be explicit going forward)

**Review by dcramer — CHANGES_REQUESTED** [review]
at the very least we'll need to adjust the webhooks plugin to not make URLs required. Not sure if any other integrations have similar behaviors.

**Review by dcramer — APPROVED** [review]

**Pull request merged**